cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/util/AwsStackNameCommonUtil.java

```diff
@@ -0,0 +1,37 @@
+package com.sequenceiq.cloudbreak.cloud.aws.common.util;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+
+@Component
+public class AwsStackNameCommonUtil {
+
+    private static final Logger LOGGER = getLogger(AwsStackNameCommonUtil.class);
+
+    private static final String INSTANCE_NAME_PATTERN = "%s-%s%d";
+
+    @Value("${cb.max.aws.resource.name.length:}")
+    private int maxResourceNameLength;
+
+    public String getInstanceName(AuthenticatedContext ac, String groupName, Long privateId) {
+        if (StringUtils.isBlank(groupName)) {
+            throw new IllegalArgumentException("Group name cannot be empty, instance name cannot be generated");
+        }
+        int lengthOfPostFix = privateId.intValue() + groupName.length();
```

Comment:
Why `privateId.intValue()`? What you are looking for instead is how many digits `privateId` will consume when rendered in decimal.

Reply:
because somewhere I need to cast to int value.
the `substring` accept int only.
`public String substring(int beginIndex, int endIndex) {`
what is the best practice here?

Reply:
As discussed privately, my comment was not about the way the casting is done, but rather the value being used here. :smile:

Reply:
We talked and the length of private id was the problem instead of the casting. I have fixed it